src/tools/ci/test_extensions

```diff
@@ -0,0 +1,300 @@
+            ['make', 'install'],
+        ],
+        'test': [
+            ['make', 'installcheck'],
+        ],
+    }
+)
+
+# add postgis
+extensions.append(
+    {
+        'name': 'postgis',
+        'download': [
+            git_clone_command + ['https://github.com/postgis/postgis.git'],
+        ],
+        'build': [
+            ['./autogen.sh'],
+            ['./configure', '--without-raster'],
+            ['make'],
+        ],
+        'test': [
+            ['make', 'check'],
+        ],
+    }
+)
+
+# add pg_partman
+extensions.append(
+    {
+        'name': 'pg_partman',
+        'shared_preload_name': 'pg_partman_bgw',
+        'download': [
+            git_clone_command
+                + ['https://github.com/pgpartman/pg_partman.git'],
+        ],
+        'build': [
+            ['make'],
+            ['make', 'install'],
+        ],
+        'test': [
+            # there is no test command,
+            # see bottom of https://github.com/pgpartman/pg_partman
+        ],
+    }
+)
+
+# add timescaledb
+# doesn't support PG 15 yet
+# extensions.append(
+#     {
+#         'name': 'timescaledb',
+#         'shared_preload_name': 'timescaledb',
+#         'download': [
+#             ['git', 'clone', 'https://github.com/timescale/timescaledb.git'],
+#         ],
+#         'build': [
+#             ['./bootstrap'],
+#             ['make', '-C', 'build'],
+#             ['make', '-C', 'build', 'install'],
+#         ],
+#         'test': [
+#             ['make', 'check'],
+#         ],
+#     }
+# )
+
+# add pgbouncer
+# since autoconf build is not working too, commented out for now
+# extensions.append(
+#     {
+#         'name': 'pgbouncer',
+#         'download': [
+#             git_clone_command
+#                 + ['https://github.com/pgbouncer/pgbouncer.git'],
+#         ],
+#         'build': [
+#             ['git', 'submodule', 'init'],
+#             ['git', 'submodule', 'update'],
+#             ['./autogen.sh'],
+#             ['./configure', '--prefix=/usr/local'],
+#             ['make'],
+#             ['make', 'install'],
+#         ],
+#         'test': [
+#             ['make', '-C', './test', 'all'],
+#             ['make', '-C', './test', 'check'],
+#         ],
+#     }
+# )
+
+# filter elements of only-run-ext list
+if only_run_ext:
+    # check if all elements of only-list exists
+    ext_names = [elem['name'] for elem in extensions]
+    if not set(only_run_ext).issubset(ext_names):
+        sys.exit('There are unsupported extensions. Supported extensions are: '
+                 + ' '.join(ext_names))
+
+    extensions = [ext for ext in extensions if ext['name'] in only_run_ext]
+
+
+def stop_and_start_postgres_server(logfile_prefix='',
+                                   shared_preload_string=''):
+    pgsql_data_path = '/tmp/pgsql/data'
+    logfile_name = '{}-logfile'.format(logfile_prefix).strip('-')
+
+    postgres_server_commands = [
+        ['pg_ctl', 'stop', '-D', pgsql_data_path],
```

Comment:
Why do you need to stop the server first? I assume that's from a prior run?

Reply:
Yes, since shared_preload_libraries could change, I am restarting the server. Also, reading logs are a bit easier since each extension will have its own log.